src/vespa/hello.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+
+import os
+import json
+from httpx import Client, Response
+from urllib.parse import urljoin
+
+
+def log_request(request):
+    print(f"> {request.method} {request.url}")
+
+def log_response(response):
+    request = response.request
+    print(f"< {request.method} {request.url} - {response.status_code}")
+    if response.status_code >= 299:
+        response.read()
+        print(f"\n{response.text}")
+
+def main():
+    # HTTPs setup
+    endpoint = os.environ["ENDPOINT"]
+    client = Client(
+        verify=False,
+        event_hooks={
+            "request": [log_request],
+            "response": [log_response, Response.raise_for_status],
+        },
+    )
+    headers = {
+        "Accept": "application/json; charset=utf-8",
+        "Content-Type": "application/json; charset=utf-8",
+    }
+
+    # index data
+    vectors = [
+        {
+            "id": "vec1",
+            "values": [0.1, 0.2, 0.3],
+            "metadata": {"genre": "drama"},
+        },
+        {
+            "id": "vec2",
+            "values": [0.2, 0.3, 0.4],
+            "metadata": {"genre": "comedy"},
+        },
+    ]
+
+    for vector in vectors:
+        data = json.dumps({"fields": vector})
+        client.post(urljoin(endpoint, "/document/v1/vector/vector/docid/" + vector["id"]), headers=headers, data=data)
```

Comment:
Does Vespa not have a bulk interface? Just curious.

Reply:
For what I found, they have a [Feed CLI](https://docs.vespa.ai/en/vespa-cli-feed.html) that wraps around the [Document API](https://docs.vespa.ai/en/document-v1-api-guide.html) (the one I'm using here). I could not find a way to pass multiple documents to index directly via API calls. Perhaps looking at how the Feed CLI does it, it is using undocumented endpoints vs just iterating through the docs and calling the Document API multiple times...